Local/Msync.py

```diff
@@ -14,41 +17,123 @@ def listening_for(self, message):
     def handle(self,message):
         try:
             target = message['metadata']['source_channel']
-            opts = {'target':target}
+            self.opts = {'target':target}
+            self.message = message
         except IndexError:
             logger.error('Could not identify message source in message: %s' \
                          % str(message))
-        arg = self.parse_args(message)
 
-        if arg == None:
-            response = self.get_current_msync()
-            self.reply(message, response, opts)
+        # A python workaround for case statements
+        dispatcher = {'getver':self._get_single_version,
+                     'olderthan':self._compare_to_version,
+                     'newerthan':self._compare_to_version,
```

Comment:
How about `olderthanoreq` and `newerthanoreq`? You could then define `olderthan` as (psuedo since I don't know python) `olderthanoreq and current != version`, similar for `newerthan`.